core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -460,6 +461,15 @@ public void initialize(URI uri, org.apache.hadoop.conf.Configuration conf) throw
     mAlluxioHeader = getScheme() + "://" + authority;
     // Set the statistics member. Use mStatistics instead of the parent class's variable.
     mStatistics = statistics;
+
+    AlluxioURI header = new AlluxioURI(mAlluxioHeader);
```

Comment:
since we only use the authority, we could parse just that part with `Authority.fromString(authority)`

Reply:
using `authority` directly won't work because the variable will get set to `"/"` if the original authority is null. `uri.getAuthority()` will work though and shouldn't add overhead because it caches the parsed authority after the first call.